### PR TITLE
Change to shorter package names for launch files

### PR DESCRIPTION
**Pull request by luca-della-vedova**
This PR follows up from the package splitting done in #78 and changes the package names from `rmf_demos_gazebo` to `rmf_demos_gz` and `rmf_demos_ignition` to `rmf_demos_ign` to reduce typing on commonly used operations (such as running the demos).

As a consequence the launching of demos is now:
`ros2 launch rmf_demos_gz office.launch.xml`
or
`ros2 launch rmf_demos_ign office.launch.xml`

Also `rmf_demos_base` is renamed back to `rmf_demos` to minimized disruption on downstream users.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -201,21 +203,21 @@ RMF can also manage fleets whose API or fleet managers only offer pause and resu
 
 #### Triple-H scenario:
 ```bash
-$ ros2 launch rmf_demos triple_H.launch.xml
-(new terminal) $ ros2 launch rmf_demos the_pedigree.launch.xml
+$ ros2 launch rmf_demos_gz triple_H.launch.xml
+(new terminal) $ ros2 launch rmf_demos_base the_pedigree.launch.xml
```
Comment (Yadunund) [inline README.md:207]:
```suggestion
(new terminal) $ ros2 launch rmf_demos the_pedigree.launch.xml
```

File: README.md [hunk 2]
```diff
@@ -201,21 +203,21 @@ RMF can also manage fleets whose API or fleet managers only offer pause and resu
...
 ```
 #### Battle Royale Scenario:
 
 ```bash
-$ ros2 launch rmf_demos battle_royale.launch.xml
-(new terminal) $ ros2 launch rmf_demos battle_go.launch.xml
+$ ros2 launch rmf_demos_gz battle_royale.launch.xml
+(new terminal) $ ros2 launch rmf_demos_base battle_go.launch.xml
```
Comment (Yadunund) [inline README.md:213]:
```suggestion
(new terminal) $ ros2 launch rmf_demos battle_go.launch.xml
```

File: README.md [hunk 3]
```diff
@@ -201,21 +203,21 @@ RMF can also manage fleets whose API or fleet managers only offer pause and resu
...
 ```
 
 #### Office Scenario:
 Note that `tinyRobot1` is a standard "full control" robot, while `tinyRobot2` "traffic light" robot.
 ```bash
-$ ros2 launch rmf_demos office_mock_traffic_light.launch.xml
-(new terminal) $ ros2 launch rmf_demos office_traffic_light_test.launch.xml
+$ ros2 launch rmf_demos_gz office_mock_traffic_light.launch.xml
+(new terminal) $ ros2 launch rmf_demos_base office_traffic_light_test.launch.xml
```
Comment (Yadunund) [inline README.md:220]:
```suggestion
(new terminal) $ ros2 launch rmf_demos office_traffic_light_test.launch.xml
```

File: rmf_demos_gz/CHANGELOG.md [hunk 4]
```diff
@@ -1,4 +1,4 @@
-## Changelog for package rmf_demos
+## Changelog for package rmf_demos_gz
```
Comment (Yadunund) [inline rmf_demos_gz/CHANGELOG.md:1]: This is a new package and shouldn't have any prior history in its changelogs (same with `rmf_demos_ign`). When we update the version/changelogs for the `galactic` release we can append the log with https://github.com/open-rmf/rmf_demos/pull/78 and this PR

**Review by Yadunund — CHANGES_REQUESTED** [review]
Thanks for the changes. Left some minor comments to address.
Another suggestion for consideration could be to set the default value of the `use_sim_time` param in the `rmf_demos/launch.*.launch.xml` files to `false` https://github.com/open-rmf/rmf_demos/blob/e2e86a3a38b2f47c28218dca26ea49dc6812cbf3/rmf_demos_base/launch/office.launch.xml#L4

This way launching these files can hypothetically run hardware testing. It won't affect the simulations since the `use_sim_time` parameter is overwritten by the the launch files in `rmf_demos_gz` and `rmf_demos_ign`.

As a follow up, I noticed that not all the nodes started in `rmf_demos/common.launch.xml` set the `use_sim_time` parameter. Maybe we can fix that in this PR?

**Comment by luca-della-vedova** [comment]
[Done](https://github.com/open-rmf/rmf_demos/pull/80/commits/50a50ae99643960ef103fb78a1d8aa450ae50bbf)! To recap:

* Fixed the readme (spurious `rmf_demos_base` lying around)
* Removed the changelogs from the `gz` and `ign` packages
* Added `use_sim_time` to all common files.
* Set `use_sim_time` to false by default in all `rmf_demos` launch files.

**Review by Yadunund — APPROVED** [review]
Awesome, thanks for all the changes!

**Pull request merged**